packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationFirebaseMessagingService.kt

```diff
@@ -41,8 +42,9 @@ class PushNotificationFirebaseMessagingService : FirebaseMessagingService() {
      * FCM registration token is initially generated so this is where you would retrieve the token.
      */
     override fun onNewToken(token: String) {
+        Log.d(TAG, "onNewToken")
         // Should initialize normally as it's initialized for the first time.
-        StreamHandlers.initStreamHandlers()
+        StreamHandlers.initStreamHandlers(false)
```

Comment:
So will `initStreamHandlers` be called on every `onNewToken` event? Can this be optimized?

Reply:
`onNewToken` doesn't fire often but sometimes fires before anything else so have to init here. Multiple `initStreamHandlers` won't do anything since there is an isInitialized flag so the initialization happens only once.